pkg/cli/admin/upgrade/status/controlplane.go

```diff
@@ -30,9 +31,52 @@ type controlPlaneStatusDisplayData struct {
 	Operators  operators
 }
 
-func assessControlPlaneStatus(cv *v1.ClusterVersion, operators []v1.ClusterOperator, at time.Time) controlPlaneStatusDisplayData {
+const (
+	unavailableWarningThreshold = 5 * time.Minute
+	unavailableErrorThreshold   = 20 * time.Minute
+	degradedWarningThreshold    = 5 * time.Minute
```

Comment:
I would suggest to wait for longer initially , example 15 mins. If the status is still degraded we can consider this as warning. In future if we feel that the warning should be given after 5 mins , we can do that. Having a longer time for the first iteration will reduce false positives.

Reply:
I think 5m is enough to smooth over useless blips and if somehting persists over 5m then I think it is worth showing..

Being more sensitive (while smoothing over blips) is better for demos and getting feedback (people will feedback on what they see, not as much on what they miss) and it is generally easier to notice that we are still too sensitive (and smooth over more) than noticing that we would like more output should be actually smoothing less.

Reply:
> Being more sensitive (while smoothing over blips) is better for demos and getting feedback (people will feedback on what they see, not as much on what they miss) and it is generally easier to notice that we are still too sensitive (and should smooth over more) than noticing that we would like more output (and should be actually smoothing less).

I agree that for demos and getting feedback/testing 5m is better. However when we GA  this to customer 15 min might be a better frequency IMO. Mostly because we do not want customers get too worked up seeing changes in every 5 mins. Also we will learn somethings when we test this out in real life scenarios. So I am fine with 5mins for now.